test/wolt_modal_sheet_test.dart

```diff
@@ -147,120 +108,172 @@ void main() {
-                },
-                child: const Text('Open sheet'),
-              );
-            },
-          ),
-        )),
-      ),
-    );
+    await tester.pumpWidget(buildSheetWithShow());
 
     await tester.tap(find.text('Open sheet'));
     await tester.pumpAndSettle();
 
-    Finder sheetMaterial = find.byType(Material).last;
+    Finder sheetMaterial = sheetPageMaterialFinder(tester);
 
     // The default modalTypeBuilder should be a bottom sheet on narrow screens.
     expect(tester.getSize(sheetMaterial), sheetPageSize);
-    expect(tester.getTopLeft(sheetMaterial), const Offset(0.0, 504.0));
-    expect(tester.getTopRight(sheetMaterial), Offset(viewSize.width, 504.0));
+    expect(tester.getTopLeft(sheetMaterial), const Offset(0.0, 514.0));
+    expect(tester.getTopRight(sheetMaterial), Offset(viewSize.width, 514.0));
   }, skip: true); // [Intended]: This is skipped due to a bug in the framework.
+
+  testWidgets('Custom WoltModalSheet.modalTypeBuilder', (tester) async {
+    await tester.pumpWidget(buildSheetWithShow(
+      modalTypeBuilder: (context) {
+        return WoltModalType.bottomSheet();
+      },
+    ));
+
+    await tester.tap(find.text('Open sheet'));
+    await tester.pumpAndSettle();
+
+    expect(tester.getSize(sheetPageMaterialFinder(tester)),
+        const Size(800.0, 86.0));
+
+    // Tap to dismiss the sheet.
+    await tester.tapAt(const Offset(50, 50));
+    await tester.pumpAndSettle();
+
+    expect(find.text('Wolt modal sheet page'), findsNothing);
+
+    await tester.pumpWidget(buildSheetWithShow(
+      modalTypeBuilder: (context) {
+        return WoltModalType.dialog();
+      },
+    ));
+
+    await tester.tap(find.text('Open sheet'));
+    await tester.pumpAndSettle();
+
+    expect(tester.getSize(sheetPageMaterialFinder(tester)),
+        const Size(524.0, 86.0));
+
+    // Tap to dismiss the sheet.
+    await tester.tapAt(const Offset(50, 50));
+    await tester.pumpAndSettle();
+
+    expect(find.text('Wolt modal sheet page'), findsNothing);
+    await tester.pumpWidget(buildSheetWithShow(
+      barrierDismissible: true,
+      modalTypeBuilder: (context) {
+        return WoltModalType.alertDialog();
+      },
+    ));
+
+    await tester.tap(find.text('Open sheet'));
+    await tester.pumpAndSettle();
+
+    expect(tester.getSize(sheetPageMaterialFinder(tester)),
+        const Size(404.0, 86.0));
+
+    // Tap to dismiss the sheet.
+    await tester.tapAt(const Offset(50, 50));
+    await tester.pumpAndSettle();
+
+    expect(find.text('Wolt modal sheet page'), findsNothing);
+
+    await tester.pumpWidget(buildSheetWithShow(
+      modalTypeBuilder: (context) {
+        return WoltModalType.sideSheet();
+      },
+    ));
+
+    await tester.tap(find.text('Open sheet'));
+    await tester.pumpAndSettle();
+
+    expect(tester.getSize(sheetPageMaterialFinder(tester)),
+        const Size(524.0, 600.0));
+  });
+
+  testWidgets('Custom WoltModalSheet.modalDecorator', (tester) async {
```

Comment:
shall we group the decorator tests?

Reply:
The goal here is to test all parameters of the constructors, eventually and not just decorators. Generally, I wouldn't group just decorators from a constructor but i can do that do here if you prefer